rasa/core/policies/rule_policy.py

```diff
@@ -661,19 +660,34 @@ def _is_rule_applicable(
         # turn_index goes back in time
         reversed_rule_states = list(reversed(self._rule_key_to_state(rule_key)))
 
-        return bool(
-            # rule is shorter than current turn index
-            turn_index >= len(reversed_rule_states)
-            # current rule and state turns are empty
-            or (not reversed_rule_states[turn_index] and not conversation_state)
-            # check that current rule turn features are present in current state turn
-            or (
-                reversed_rule_states[turn_index]
-                and conversation_state
-                and self._does_rule_match_state(
-                    reversed_rule_states[turn_index], conversation_state
-                )
-            )
+        # the rule must be applicable because we got (without any applicability issues)
+        # further in the conversation history than the rule's length
+        if turn_index >= len(reversed_rule_states):
+            return True
+
+        # a state has previous action if and only if it is not a conversation start
+        # state
+        current_previous_action = conversation_state.get(PREVIOUS_ACTION)
+        rule_previous_action = reversed_rule_states[turn_index].get(PREVIOUS_ACTION)
+
+        # current rule state is a conversation starter (due to conversation_start: true)
+        # but current conversation state is not.
+        # or
+        # current conversation state is a starter (due to initial_value set for a slot)
+        # but current rule state is not a starter
```

Comment:
@Ghostvv could you look at this case? I don't see why it would make the rule not applicable. But I also don't think this case can every happen with our current implementation... Either way, my brain hurts from trying to understand this and trying to write a test case that would point out the possible mistake here.

Reply:
not sure what you mean, but I updated your code, it should be equivalent to the bool now

Reply:
Thanks for the changes!